### DISCUSSION TITLE
开源代码现已支持编译、运行！

### DISCUSSION THREAD

**Discussion by LTCatt** [Meloong-Git/PCL#3910] [公告]
本存储库中的代码现已经可以在 Visual Studio 中编译、运行，点击 `Code → Open with Visual Studio` 即可。

![image](https://github.com/Hex-Dragon/PCL2/assets/40348278/34d6f0a3-8c38-45cb-bd28-1c6606b7169c)

可编译开源版本仅不包含以下特性：
- **CurseForge 查询：** 它需要私有的 API Key。这会导致 Mod 和整合包的下载、更新功能无法访问 CurseForge。如果你有一个 API Key，可以把它添加到 SecretHeadersSign 方法中以恢复这些功能。
- **更新与联网通知：** 避免滥用隐患。
- **主题切换：** 这是需要赞助解锁的纪念性质的功能，别让赞助者太伤心啦……
- **百宝箱：** 开发早期往里面塞了些开发工具，整理起来太麻烦了……

**Comment by AlphaAntivoid** [comment]
非常好的开源软件，使我的Minecraft旋转👍🏻

**Comment by wuliaodexiaoluo** [comment]
激动人心的时刻终于来了
有没有开发者模式？
能不能 action 编译

**Reply by allMagicNB** [reply]
:trollface:https://github.com/allMagicNB/PCL2/actions

**Comment by LTCatt** [comment]
修改了一下协议……之前的协议按理说是不允许将魔改版开源的，现在明确可以开源了 =。=

如果你希望将魔改版开源，请遵循：
（一）给出适当的说明，表示这是第三方内容。
（二）给出适当的署名，且提供指向本存储库的链接。
（三）不得以任何方式暗示开发者为您或您的使用背书。
（四）使用[指南文件](https://github.com/Hex-Dragon/PCL2/blob/main/LICENCE)作为协议，不得附加额外条款。
（五）不得分发编译后的可执行文件。

**Reply by Silverteal** [reply]
也就是说，满足条件的情况下，修改的版本**只发布源代码**是可以的？

**Reply by burningtnt** [reply]
All rights reserved. 这从来不是一个开源项目

**Reply by Silverteal** [reply]
> All rights reserved. 这从来不是一个开源项目

有时候，开源有着一个按字面意思理解的解释——源码可见（open to public），虽然并不常用，但也不能说就完全不是

**Reply by LTCatt** [reply]
理论上这是“开源的”，但不是“开源软件”。我按照个人习惯这样说的，如果造成不适非常抱歉（
满足条件的情况下可以发布修改版源码，文件里写了具体的情况。

**Reply by Deep-Dark-Forest** [reply]
@LTCatt 如果我不提供 exe 文件但是提供了详细的编译源代码的教程呢？

**Reply by Silverteal** [reply]
> @LTCatt 如果我不提供 exe 文件但是提供了详细的编译源代码的教程呢？

问得很多余，各种意义上的。

**Reply by burningtnt** [reply]
> > @LTCatt 如果我不提供 exe 文件但是提供了详细的编译源代码的教程呢？
> 
> 问得很多余，各种意义上的。

> 不得分发编译后的可执行文件。

如果我分发另一个 Wrapper EXE（完全与 PCL2 无关的程序），他会自动下载修改后的源码并编译呢？

**Reply by allMagicNB** [reply]
这就是 LICENSE 自己写的弊端

**Reply by Silverteal** [reply]
> > > @LTCatt 如果我不提供 exe 文件但是提供了详细的编译源代码的教程呢？
> > 
> > 
> > 问得很多余，各种意义上的。
> 
> > 不得分发编译后的可执行文件。
> 
> 如果我分发另一个 Wrapper EXE（完全与 PCL2 无关的程序），他会自动下载修改后的源码并编译呢？

参照Forge，我想没有什么问题.....:)
